scene/gui/scroll_container.cpp

```diff
@@ -602,6 +604,48 @@ void ScrollContainer::update_scrollbars() {
 	callable_mp(this, &ScrollContainer::_update_scrollbar_position).call_deferred();
 }
 
+void ScrollContainer::update_scroll_hints() {
+	Size2 size = get_size();
+	Rect2 margins = _get_margins();
+	Size2 scroll_size = size - margins.position + margins.size;
+
+	float v_scroll_value = v_scroll->get_value();
+	bool v_scroll_below_max = v_scroll_value < (largest_child_min_size.height - scroll_size.height - 1);
+	bool show_vertical_hints = v_scroll_value > 1 || v_scroll_below_max;
+
+	float h_scroll_value = h_scroll->get_value();
+	bool h_scroll_below_max = h_scroll_value < (largest_child_min_size.width - scroll_size.width - 1);
+	bool show_horizontal_hints = h_scroll_value > 1 || h_scroll_below_max;
+
+	if (show_vertical_hints) {
+		scroll_hint_top_left->set_texture(theme_cache.scroll_hint_vertical);
+		scroll_hint_top_left->set_visible(!show_horizontal_hints && (scroll_hint_mode == SCROLL_HINT_MODE_ALL || scroll_hint_mode == SCROLL_HINT_MODE_TOP_AND_LEFT) && v_scroll_value > 1);
+		scroll_hint_top_left->set_offsets_preset(PRESET_TOP_WIDE, LayoutPresetMode::PRESET_MODE_MINSIZE);
+
+		scroll_hint_bottom_right->set_flip_h(false);
+		scroll_hint_bottom_right->set_flip_v(true);
+		scroll_hint_bottom_right->set_texture(theme_cache.scroll_hint_vertical);
+		scroll_hint_bottom_right->set_visible(!show_horizontal_hints && (scroll_hint_mode == SCROLL_HINT_MODE_ALL || scroll_hint_mode == SCROLL_HINT_MODE_BOTTOM_AND_RIGHT) && v_scroll_below_max);
+		scroll_hint_bottom_right->set_offsets_preset(PRESET_BOTTOM_WIDE, LayoutPresetMode::PRESET_MODE_MINSIZE);
+	} else {
+		bool rtl = is_layout_rtl();
+
+		scroll_hint_top_left->set_texture(theme_cache.scroll_hint_horizontal);
+		scroll_hint_top_left->set_visible(!show_vertical_hints && (scroll_hint_mode == SCROLL_HINT_MODE_ALL || (rtl ? scroll_hint_mode == SCROLL_HINT_MODE_BOTTOM_AND_RIGHT : scroll_hint_mode == SCROLL_HINT_MODE_TOP_AND_LEFT)) && h_scroll_value > 1);
+		scroll_hint_top_left->set_offsets_preset(rtl ? PRESET_RIGHT_WIDE : PRESET_LEFT_WIDE, LayoutPresetMode::PRESET_MODE_MINSIZE);
+		if (rtl) {
+			// TODO: Find out why this is even necessary.
+			scroll_hint_top_left->set_position(Point2(0, scroll_hint_top_left->get_position().y));
+		}
```

Comment:
I have no clue why, but this is necessary for this hint to be positioned properly.

Reply:
I don't see the problem. Can you show a screenshot of the wrong position?